src/injected/analyzer-state-update-handler.ts

```diff
@@ -41,19 +35,27 @@ export class AnalyzerStateUpdateHandler {
         prevState: VisualizationStoreData,
         currState: VisualizationStoreData,
     ): void {
-        const types = EnumHelper.getNumericValues<VisualizationType>(VisualizationType);
-        types.forEach(visualizationType => {
-            if (prevState != null) {
-                const configuration =
-                    this.visualizationConfigurationFactory.getConfiguration(visualizationType);
-                const keys = this.getTestKeysFromConfiguration(configuration, currState);
-                keys.forEach(testKey => {
-                    if (this.isTestTerminated(configuration, prevState, currState, testKey)) {
-                        this.teardown(configuration.getIdentifier(testKey));
-                    }
-                });
-            }
-        });
+        if (prevState == null) {
+            return;
+        }
+
+        this.visualizationConfigurationFactory.forEachConfig(
+            (configuration, type, requirementConfig) => {
```

Comment:
```suggestion
           (configuration: VisualizationConfiguration, type: VisualizationType, requirementConfig: Requirement) => {
```

Explicit types increase readability.